### PR TITLE
728: improve rose suite-run migration docs

### LINKED ISSUE #513: major changes: document remote-init change from rose suite-run
* `rose suite-run` installed everywhere *before* startup.
* `cylc play` installs where required as required.

This creates an issue for tasks which install on remote platforms as the run dir structure and and configured symlinks will not be created until the first remote task runs.

The solution is to add a remote task into the graph which runs *before* the local task (which installs onto the remote) e.g:

```
install_remote => install_stuff_on_remote => do_something_on_remote
```

Note: This is made a lot more elegant by https://github.com/cylc/cylc-flow/pull/5036

Two-stage `fcm_make` builds are a likely candidate of migration pain e.g. https://cylc.discourse.group/t/cylc-8-2-step-fcm-make-and-symlink-dirs/515

### PR DESCRIPTION
**Pull request by oliver-sanders**
Built on #516.

Closes #513 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.

### CODE REVIEW

File: src/7-to-8/major-changes/cylc-install.rst [hunk 1]
```diff
@@ -112,40 +113,64 @@ use :cylc:conf:`flow.cylc[scheduler]install`.
...
+   With Rose 2019 / ``rose suite-run``, files were installed on remote platforms
+   *before the workflow started* running.
+
+   With Rose 2 / ``cylc install``, files are installed on remote platforms just
+   *before the first task* runs on that platform.
 
-   :ref:`Rose Stem` documentation.
+   Rose used to install the entire workflow :term:`run directory` to remote
+   platforms. It now only installs configured directories for efficiency.
+   See `Remote Installation` above for details.
```
Comment (wxtim) [inline src/7-to-8/major-changes/cylc-install.rst:176]: Either 

```suggestion
   See :ref:`Remote Installation <728.remote-install>` above for details.
```
or 
```suggestion
   See ``Remote Installation`` above for details.
```

**Review by wxtim — CHANGES_REQUESTED** [review]
- [x] Read, Reasonable
- [ ] Builds correctly

File: src/7-to-8/major-changes/cylc-install.rst [hunk 2]
```diff
@@ -112,40 +113,64 @@ use :cylc:conf:`flow.cylc[scheduler]install`.
...
+Roses directory:
+   By default ``cylc install`` looks for workflows in ``~/cylc-src``, you
+   you may want to add ``~/roses`` to the list of
+   :cylc:conf:`global.cylc[install]source dirs`.
+Remote Installation:
+   With Rose 2019 / ``rose suite-run``, files were installed on remote platforms
+   *before the workflow started* running.
+
+   With Rose 2 / ``cylc install``, files are installed on remote platforms just
+   *before the first task* runs on that platform.
```
Comment (MetRonnie) [inline src/7-to-8/major-changes/cylc-install.rst:172]:
```suggestion
   before the *first task* runs on that platform.
```

File: src/7-to-8/major-changes/cylc-install.rst (hunk 1 above)
Comment (MetRonnie) [reply src/7-to-8/major-changes/cylc-install.rst:176]: I say the first one

**Review by MetRonnie — COMMENTED** [review]
Couple of suggestions

File: src/7-to-8/major-changes/cylc-install.rst (hunk 1 above)
Comment (oliver-sanders) [reply src/7-to-8/major-changes/cylc-install.rst:176]: Ah, needs a trailing `_`.

Comment (oliver-sanders) [inline src/7-to-8/major-changes/cylc-install.rst:176]:
```suggestion
   See `Remote Installation`_ above for details.
```

Comment (MetRonnie) [reply src/7-to-8/major-changes/cylc-install.rst:176]: > src/7-to-8/major-changes/cylc-install.rst:174:cylc:conf reference target not found: Remote Installation

**Review by MetRonnie — APPROVED** [review]

**Review by wxtim — APPROVED** [review]

**Pull request merged**